versioned/tiered/dynamodb/src/main/java/com/dremio/nessie/versioned/store/dynamo/DynamoRef.java

```diff
@@ -60,125 +60,173 @@
   static final String METADATA = "metadata";
   static final String KEY_LIST = "keys";
 
+  private boolean tag;
```

Comment:
please avoid using multiple parallel booleans. Best in these situations to use an enum such as INIT, TAG, BRANCH